Engine/Results/SymbolCapacity.cs

```diff
@@ -0,0 +1,279 @@
+ *
+*/
+
+using System;
+using System.Linq;
+using QuantConnect.Data.Market;
+using QuantConnect.Interfaces;
+using QuantConnect.Orders;
+using QuantConnect.Securities;
+
+namespace QuantConnect.Lean.Engine.Results
+{
+    /// <summary>
+    /// Per-symbol capacity estimations, tightly coupled with the <see cref="CapacityEstimate"/> class.
+    /// </summary>
+    internal class SymbolCapacity
+    {
+        /// <summary>
+        /// An estimate of how much volume the FX market trades per minute
+        /// </summary>
+        private const decimal _forexMinuteVolume = 25000000m;
+        private const decimal _fastTradingVolumeScalingFactor = 2m;
+
+        private readonly IAlgorithm _algorithm;
+        private readonly Symbol _symbol;
+        private readonly SecurityHolding _holding;
+
+        private decimal _previousVolume;
+        private DateTime? _previousTime;
+
+        private decimal _averageDollarVolume;
+        private decimal _marketCapacityDollarVolume;
+        private bool _resetMarketCapacityDollarVolume;
+        private decimal _fastTradingVolumeDiscountFactor;
+        private OrderEvent _previousOrderEvent;
+        private TradeBar _previousBar;
+        private Resolution _resolution;
+
+        /// <summary>
+        /// The Symbol's Security
+        /// </summary>
+        public Security Security { get; }
+
+        private decimal _resolutionScaleFactor
+        {
+            get
+            {
+                switch (_resolution)
+                {
+                    case Resolution.Daily:
+                        return 0.02m;
+
+                    case Resolution.Hour:
+                        return 0.05m;
+
+                    case Resolution.Minute:
+                        return 0.20m;
+
+                    case Resolution.Tick:
+                    case Resolution.Second:
+                        return 0.50m;
+
+                    default:
+                        return 1m;
+                }
+            }
+        }
+
+        /// <summary>
+        /// The absolute dollar volume we've traded
+        /// </summary>
+        public decimal SaleVolume { get; private set; }
+
+        /// <summary>
+        /// Market capacity dollar volume, i.e. the capacity the market is able to provide for this Symbol
+        /// </summary>
+        public decimal MarketCapacityDollarVolume => _marketCapacityDollarVolume * _resolutionScaleFactor;
+
+        /// <summary>
+        /// Creates a new SymbolCapacity object, capable of determining market capacity for a Symbol
+        /// </summary>
+        /// <param name="algorithm"></param>
+        /// <param name="symbol"></param>
+        public SymbolCapacity(IAlgorithm algorithm, Symbol symbol)
+        {
+            _algorithm = algorithm;
+            Security = _algorithm.Securities[symbol];
+            _symbol = symbol;
+            _holding = _algorithm.Portfolio[_symbol];
+
+            var resolution = Security.Subscriptions
```

Comment:
`_resolution` might not be static, meaning securities could be added with different resolutions.
Let's replace usage of `Security.Subscriptions` for `algorithm.SubscriptionManager.SubscriptionDataConfigService.GetSubscriptionDataConfigs(symbol)` to get the subscriptions, the first isn't intended to be used by the engine really.